docs/dev_guide/contents/map_rotate_custom.rst

```diff
@@ -0,0 +1,84 @@
+.. _map_rotate_custom:
+
+****************************
+Adding a new rotation method
+****************************
+
+If you want to add a new rotation method that can be used by `sunpy.map.GenericMap.rotate`, you can do the following:
+
+.. code-block:: python
+
+    from sunpy.image.transform import add_rotation_function
+
+    @add_rotation_function("my_rotate", handles_clipping=False, handles_image_nans=False, handles_nan_missing=False)
+    def _rotation_mine(image, matrix, shift, order, missing, clip):
+        # Rotation code goes here
+        return rotated_image
+
+The best explanation of what the arguments mean is:
+
+.. autofunction:: sunpy.image.transform.add_rotation_function
+   :noindex:
+
+Then you can do:
+
+.. code-block:: python
+
+    from sunpy.map import Map
+    from sunpy.data import sample
+
+    hmi_map = sunpy.map.Map(sample.HMI_LOS_IMAGE)
+
+    rot_map = hmi_map.rotate(order=3, recenter=True, method="my_rotate")
+
+Below is a non-functioning OpenCV2 rotation method as an example:
```

Comment:
Why non-funcitoning? Can we make it functioning without much effort?

Reply:
It lacks proper handling of the shift (based off the comments from the original PR) and someone who understands the differences in the affine transform in openCV2 could make it functioning with some effort.